### PR TITLE
IS-537: Add listener interface to reward calculator proxy

### PR DESCRIPTION
**Pull request by eunsoo-icon**
- Can add callback function for CALCULATE response message with RewardCalcProxy()
- Can handle VERSION message with version_handler()@RewardCalcProxy
- Add 0.1 sec timeout to the CLAIM, QUERY, COMMIT_BLOCK message IPC communication
    - raise TimeoutException

### CODE REVIEW

File: iconservice/iiss/ipc/reward_calc_proxy.py [hunk 1]
```diff
@@ -35,23 +36,27 @@ class RewardCalcProxy(object):
...
 
         Logger.debug(tag=_TAG, msg="__init__() end")
 
     def open(self, sock_path: str, iiss_db_path: str):
         Logger.debug(tag=_TAG, msg="open() start")
 
         self._loop = asyncio.get_event_loop()
-        self._message_queue = MessageQueue(self._loop)
+        self._message_queue = MessageQueue(loop=self._loop,
+                                           notify_message=VersionResponse,
```
Comment (leeheonseung) [inline iconservice/iiss/ipc/reward_calc_proxy.py]: I'm not sure it is okey to ignore type mismatch. becuase that param's type is tuple, but you put the response instance.

Comment (eunsoo-icon) [reply iconservice/iiss/ipc/reward_calc_proxy.py]: how about the changes below?
```suggestion
                                           notify_message=(VersionResponse,),
```

File: iconservice/iiss/ipc/reward_calc_proxy.py [hunk 2]
```diff
@@ -35,23 +36,27 @@ class RewardCalcProxy(object):
...
     """
 
-    def __init__(self):
+    IPC_TIMEOUT = 0.1
+
+    def __init__(self, calc_callback: Callable[[tuple], Any] = None):
         Logger.debug(tag=_TAG, msg="__init__() start")
 
         self._loop = None
         self._message_queue = None
```
Comment (leeheonseung) [inline iconservice/iiss/ipc/reward_calc_proxy.py]: redundant member

File: iconservice/iiss/ipc/reward_calc_proxy.py (hunk 1 above)
Comment (leeheonseung) [reply iconservice/iiss/ipc/reward_calc_proxy.py]: okay, reasonable

File: iconservice/iiss/ipc/reward_calc_proxy.py [hunk 3]
```diff
@@ -127,21 +136,21 @@ def calculate(self, db_path: str, block_height: int):
...
     def on_calculate_done(self, future: asyncio.Future):
         Logger.debug(tag=_TAG, msg="on_calculate_done() start")
 
         response: 'CalculateResponse' = future.result()
-        self._calculation_result = \
-            response.success, response.block_height, response.state_hash
+        calculation_result = response.success, response.block_height, response.iscore, response.state_hash
+
+        if self._calculation_callback is not None:
+            self._calculation_callback(calculation_result)
```
[the remark this answers is not in the document]
Comment (eunsoo-icon) [reply iconservice/iiss/ipc/reward_calc_proxy.py]: Use the CalculateResponse instance as a parameter of the calculate callback function

**Pull request merged**